--additional-namespaces has problem with RBACs sealed-secrets pod throwing error

Guide to implementiing additional namespaces: [How to use one controller for a subset of namespaces](https://github.com/bitnami-labs/sealed-secrets/issues/891)
**Which component**:
Controller version : v0.18.0

**Describe the bug**
Encountered an issue after Adding - --additional-namespaces=**nsb** in values.yaml under args
```
args: 
  - --additional-namespaces=nsb
```

```
022/07/26 14:38:30 Starting informer for namespace: kube-system
panic: namespaces "nsb" is forbidden: User "system:serviceaccount:kube-system:sealed-secrets" cannot get resource "namespaces" in API group "" in the namespace "nsb"
goroutine 1 [running]:
main.main()
	/home/runner/work/sealed-secrets/sealed-secrets/cmd/controller/main.go:292 +0x174
```


**To Reproduce**
Steps to reproduce the behavior:

1. add in values.yaml
```
args: 
  - --additional-namespaces=nsb
```
2. install the helm chart helm install sealed-secret ./sealed-secret
3. Check sealed-secret pod logs and an error will appear 
```
022/07/26 14:38:30 Starting informer for namespace: kube-system
panic: namespaces "nsb" is forbidden: User "system:serviceaccount:kube-system:sealed-secrets" cannot get resource "namespaces" in API group "" in the namespace "nsb"
goroutine 1 [running]:
main.main()
	/home/runner/work/sealed-secrets/sealed-secrets/cmd/controller/main.go:292 +0x174
```

**Expected behavior**
Controller should be able to whitelist this the namespace provided on the args.

**Version of Kubernetes**:

- Output of `kubectl version`:

```
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"092fbfbf53427de67cac1e9fa54aaa09a28371d7", GitTreeState:"clean", BuildDate:"2021-06-16T12:53:14Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**
Add any other context about the problem here.

--- a/controller.jsonnet
+++ b/controller.jsonnet
@@ -30,6 +30,11 @@ controller {
         resources: ['events'],
         verbs: ['create', 'patch'],
       },
+      {
+        apiGroups: [''],
+        resources: ['namespaces'],
+        verbs: ['get'],
+      },
     ],
   },
 

--- a/helm/sealed-secrets/Chart.yaml
+++ b/helm/sealed-secrets/Chart.yaml
@@ -14,4 +14,4 @@ maintainers:
     url: https://github.com/bitnami-labs/sealed-secrets
 name: sealed-secrets
 type: application
-version: 2.4.0
+version: 2.4.1

--- a/helm/sealed-secrets/Chart.yaml
+++ b/helm/sealed-secrets/Chart.yaml
@@ -14,4 +14,4 @@ maintainers:
     url: https://github.com/bitnami-labs/sealed-secrets
 name: sealed-secrets
 type: application
-version: 2.4.1
+version: 2.5.0

--- a/helm/sealed-secrets/README.md
+++ b/helm/sealed-secrets/README.md
@@ -93,6 +93,7 @@ The command removes all the Kubernetes components associated with the chart and 
 | `secretName`                                      | The name of an existing TLS secret containing the key used to encrypt secrets        | `sealed-secrets-key`                |
 | `updateStatus`                                    | Specifies whether the Sealed Secrets controller should update the status subresource | `true`                              |
 | `keyrenewperiod`                                  | Specifies key renewal period. Default 30 days                                        | `""`                                |
+| `additionalNamespaces`                            | List of namespaces used to manage the Sealed Secrets                                 | `[]`                                |
 | `command`                                         | Override default container command                                                   | `[]`                                |
 | `args`                                            | Override default container args                                                      | `[]`                                |
 | `livenessProbe.enabled`                           | Enable livenessProbe on Sealed Secret containers                                     | `true`                              |

--- a/helm/sealed-secrets/templates/cluster-role.yaml
+++ b/helm/sealed-secrets/templates/cluster-role.yaml
@@ -39,4 +39,14 @@ rules:
     verbs:
       - create
       - patch
+  {{- if .Values.additionalNamespaces }}
+  - apiGroups:
+      - ""
+    resources:
+      - namespaces
+    resourceNames:
+      {{- include "sealed-secrets.render" (dict "value" .Values.additionalNamespaces "context" $) | nindent 6 }}
+    verbs:
+      - get
+  {{- end }}
 {{ end }}

--- a/helm/sealed-secrets/templates/deployment.yaml
+++ b/helm/sealed-secrets/templates/deployment.yaml
@@ -80,7 +80,7 @@ spec:
             - {{ .Values.secretName | quote }}
             {{- if .Values.additionalNamespaces }}
             - --additional-namespaces
-            - {{ join "," .Values.additionalNamespaces | quote}}
+            - {{ join "," .Values.additionalNamespaces | quote }}
             {{- end }}
           {{- end }}
           image: {{ printf "%s/%s:%s" .Values.image.registry .Values.image.repository .Values.image.tag }}

--- a/helm/sealed-secrets/templates/deployment.yaml
+++ b/helm/sealed-secrets/templates/deployment.yaml
@@ -78,6 +78,10 @@ spec:
             {{- end }}
             - --key-prefix
             - {{ .Values.secretName | quote }}
+            {{- if .Values.additionalNamespaces }}
+            - --additional-namespaces
+            - {{ join "," .Values.additionalNamespaces | quote}}
+            {{- end }}
           {{- end }}
           image: {{ printf "%s/%s:%s" .Values.image.registry .Values.image.repository .Values.image.tag }}
           imagePullPolicy: {{ .Values.image.pullPolicy }}

--- a/helm/sealed-secrets/values.yaml
+++ b/helm/sealed-secrets/values.yaml
@@ -67,6 +67,9 @@ rateLimit:
 ## @param rateLimitBurst Number of requests allowed to exceed the rate limit per second for verify endpoint
 ##
 rateLimitBurst: 
+## @param additionalNamespaces List of namespaces used to manage the Sealed Secrets
+##
+additionalNamespaces: []
 ## @param command Override default container command
 ##
 command: []

